pkgs/development/python-modules/betterproto/default.nix

```diff
@@ -0,0 +1,66 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, pythonOlder
+, poetry-core
+, grpclib
+, python-dateutil
+, dataclasses
+, black
+, jinja2
+, isort
+, callPackage
+}:
+let
+  version = "2.0.0b5";
+in
+buildPythonPackage {
+  pname = "betterproto";
+  inherit version;
+  format = "pyproject";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "danielgtaylor";
+    repo = "python-betterproto";
+    rev = "v${version}";
+    sha256 = "sha256-XyXdpo3Yo4aO1favMWC7i9utz4fNDbKbsnYXJW0b7Gc=";
+  };
+
+  nativeBuildInputs = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    grpclib
+    python-dateutil
+  ] ++ lib.optional (pythonOlder "3.7") [
+    dataclasses
+  ];
+
+  passthru.optional-dependencies.compiler = [
+    black
+    jinja2
+    isort
+  ];
+
+  pythonImportsCheck = [ "betterproto" ];
+
+  # Most of the tests require code generation with the
+  # protoc-gen-python_betterproto script defined by the package and thus not
```

Comment:
The package is installed before running tests so that script should be available.

Reply:
The package is installed but the script is not added to PATH as far as I can tell. Protoc expects plugins as random executables with a bespoke name. Thus the code generation fails when the script is not in PATH.

Reply:
Then we can probably do something like

```nix
preCheck = ''
  export PATH=$PATH:$out/bin
'';
```

I am trying to avoid the extra complexity when we move the tests into an extra file.

Reply:
Ahh thank you, that was the missing piece!